woof/code/snake-code.html

```diff
@@ -0,0 +1,241 @@
+    forever(() => {
+     if(mouseDown === true && mouseX > 0) {
+       snakeHead.turnRight(5);
+     }
+     if (mouseDown === true && mouseX < 0) {
+       snakeHead.turnLeft(5);
+     }
+    })
+    
+    // Creates variable "score" (set to 0), which is then displayed through text variable "scoreText"
+    var score = 0;
+    var scoreText = new Text({text: () => "Score: " + score, x: 0, y: maxY - 50, size: 30})
+    
+    
+    //Food Cloning
+    var foodArray = []  // Creates array "foodArray" to store clones
+    every(2, "seconds", () => {  // Creates a new clone every 2 seconds
+      var food = new Circle({radius: 5, color: "purple"})   // Creates variable "food" that will be cloned
+      food.x = randomX()
+      food.y = randomY()            // Sets the clone's xy coordinates to a random spot along the screen 
+      foodArray.push(food)     // Pushes each "food" clone into the array "foodArray"
+    });
+    
+    
+    //Increases "score" by one if "snakeHead" touches anything in the "foodArray" array
+    forever(() => {
+      foodArray.forEach(food => {
+        if (food.touching(snakeHead)) {
+          score++
+        }
+      })
+    })
+    
+    //Body creation
+    var bodyArray = []  // Creates array "bodyArray" to store clones
+    forever(() => {
+      foodArray.forEach(food => {
+        if(food.touching(snakeHead)) {
+          var body = new Circle({radius: 5, x: snakeHead.x, y: snakeHead.y, color: "green"}); // Creates a new "body" clone each time "snakeHead" touches a "food" clone
+          bodyArray.push(body);  // Pushes "body" to array "bodyArray"
+          food.delete()  // Deletes "food" clones when "snakeHead" touches them
+        }
+      }) 
+    })
+    
+    //Index for body following the previously made piece
+    var follow_distance  // Creates variable "follow_distance"
+    
+    forever(() => {
+      bodyArray.forEach((piece, index) => {  // Creates an index for the array "bodyArray"
+        var following;  // Creates variable "following" 
+        if (index === 0){
+          following = snakeHead  // Makes it so that if "index" equals 0, then "following" equals "snakeHead" (the pieces follow "snakeHead")
+          
+        } else {
+          following = bodyArray[index - 1] // Makes it so that if "index" is at anything else, then "following" equals "bodyArray"s current "index" minus 1 (the pieces follow the previous piece)
+        }
+        if(piece.distanceTo(following) > 27) {  // Sets up code for the situation when a "piece"s distance to whatever "following" is is greater than 27
+          piece.pointTowards(following);  // Makes the "piece" point towards whatever "following" currently is
+          piece.move(difficulty)     // Makes each piece move at the rate of "difficulty"
+        }
+      })
+    })  
+    
+    // Code for "snakeHead" touching the "body"
+    var touchable_bodyArray = []  // Creates array "touchable_bodyArray"
```

Comment:
The problem with the original code was as soon as the snake touched the food, the game would end, since snakeHead was touching its body automatically. The touchable_bodyArray delays the interaction, so that the body piece gets the chance to trail behind the snakeHead before touching it ends the game.